photon-api/src/main/scala/com/linkedin/photon/ml/data/CoordinateDataConfiguration.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data
+
+import com.linkedin.photon.ml.Types._
+import com.linkedin.photon.ml.projector.ProjectorType
+
+/**
+ * Generic trait for a configuration to define a coordinate data set.
+ */
+protected[ml] sealed trait CoordinateDataConfiguration {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

A new trait to unify the old `FixedEffectDataConfiguration` and `RandomEffectDataConfiguration` classes. Move the two aforementioned classes into the same file, and get rid of their parsing logic.